server/app.js

```diff
@@ -0,0 +1,71 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const morgan = require('morgan');
+const db = require('../database/index');
+
+
+const app = express();
+app.use(bodyParser.json());
+app.use(morgan('dev'));
+
+let staticPath = __dirname + '/../public';
+app.use('/books/:id', express.static(staticPath));
+
+//get initinal details
+app.get('/books/:id/details', (req, res) => {
+  const id = req.params.id;
+
+  db.getDetails(id)
+    .then(results => {
+      let details = results[0][0];
+
+      if (!details) {
+        res.status(404).send('no data @ specified id')
+      } else {
+        res.send(details);
+      }
+    })
```

Comment:
you can try to add a catch method to your promises in case the request gets stuck for any reason, so this will help prevent a connection timeout error

Reply:
good point! I will add that right away :)